src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -4003,18 +4005,17 @@ private bool IgnoreUserDefinedSpanConversions(TypeSymbol? source, TypeSymbol? ta
                 return false;
             }
 
-            // PROTOTYPE: Is it fine that this check is not performed when Compilation is null?
-            return Compilation?.IsFeatureEnabled(MessageID.IDS_FeatureFirstClassSpan) == true &&
-                (ignoreUserDefinedSpanConversionsInOneDirection(Compilation, source, target) ||
-                ignoreUserDefinedSpanConversionsInOneDirection(Compilation, target, source));
+            // Note: when Compilation is null, we assume latest LangVersion.
+            return Compilation?.IsFeatureEnabled(MessageID.IDS_FeatureFirstClassSpan) != false &&
```

Comment:
This check is repeated several times, with the same comment. Consider creating a private `IsFirstClassSpansEnabled` property so this is captured in one location.